lib/server.go

```diff
@@ -2488,27 +2511,36 @@ func (srv *Server) _startAddressRelayer() {
 								Services:  (ServiceFlag)(bestAddress.Services),
 							},
 						},
-					}, false)
+					}); err != nil {
+						glog.Errorf("Server.startAddressRelayer: Problem sending "+
+							"MsgDeSoAddr to RemoteNode (id= %v): %v", rn.GetId(), err)
+					}
 				}
 			}
 		}
 
-		glog.V(2).Infof("Server.Start._startAddressRelayer: Seeing if there are addrs to relay...")
+		glog.V(2).Infof("Server.startAddressRelayer: Seeing if there are addrs to relay...")
 		// Broadcast the addrs we have to all of our peers.
-		addrsToBroadcast := srv._getAddrsToBroadcast()
+		addrsToBroadcast := srv.getAddrsToBroadcast()
 		if len(addrsToBroadcast) == 0 {
-			glog.V(2).Infof("Server.Start._startAddressRelayer: No addrs to relay.")
+			glog.V(2).Infof("Server.startAddressRelayer: No addrs to relay.")
 			time.Sleep(AddrRelayIntervalSeconds * time.Second)
 			continue
 		}
 
-		glog.V(2).Infof("Server.Start._startAddressRelayer: Found %d addrs to "+
+		glog.V(2).Infof("Server.startAddressRelayer: Found %d addrs to "+
 			"relay: %v", len(addrsToBroadcast), spew.Sdump(addrsToBroadcast))
 		// Iterate over all our peers and broadcast the addrs to all of them.
-		for _, pp := range srv.cmgr.GetAllPeers() {
-			pp.AddDeSoMessage(&MsgDeSoAddr{
+		for _, rn := range remoteNodes {
+			if !rn.IsHandshakeCompleted() {
+				continue
+			}
+			if err := rn.SendMessage(&MsgDeSoAddr{
```

Comment:
nit, clean up nesting for readability
```suggestion
			addrMsg := MsgDeSoAddr{
```